hamlet/theses/views.py

```diff
@@ -141,8 +141,8 @@ class UploadRecommendationView(FormView):
     def _get_similar_documents(self, doc):
         # Split document into a list of words and infer the docvec.
         bag_of_words = []
-        for chunk in doc.chunks():
-            bag_of_words.extend(str(chunk).strip().split(' '))
+        for line in doc:
+            bag_of_words.extend(line.decode('utf-8').strip().split())
```

Comment:
Do we have any particular guarantee that this *is* UTF-8?

Reply:
No, since they are uploading whatever, it could be whatever. We'd need to use chardet or something to have any confidence in our encoding assumptions. We should probably a new ticket for this.

Reply:
Though, in the absence of any other information, UTF-8 is as good a guess as any.

Reply:
A new ticket to confirm encoding is 🌈 . I used UnicodeDammit for that in solenoid and it was great.